src/components/PeopleFilters/PeopleFilters.tsx

```diff
@@ -0,0 +1,150 @@
+import { Link, useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+
+export const PeopleFilters = () => {
+  const centuriesList = [16, 17, 18, 19, 20];
+  const [searchParams, setSearchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || 'all';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const handleQuery = (ch: string) => {
+    const params = new URLSearchParams(searchParams);
+
+    params.set('query', ch);
+
+    const check = { query: ch };
+
+    setSearchParams(params);
+    if (!check.query) {
+      params.delete('query');
+      setSearchParams(params);
+    }
+  };
+
+  const handleSex = (gender: string) => {
+    const params = new URLSearchParams(searchParams);
+
+    params.set('sex', gender);
+    if (gender === 'all') {
+      params.delete('sex');
+    }
+
+    return params.toString();
+  };
+
+  const handleCenturies = (century: string) => {
+    const params = new URLSearchParams(searchParams);
+
+    const newCenturies = centuries.includes(century.toString())
+      ? centuries.filter(cent => cent !== century.toString())
+      : [...centuries, century];
+
+    params.delete('centuries');
+    if (century !== '') {
+      newCenturies.forEach(cen => params.append('centuries', cen.toString()));
+    }
+
+    return params.toString();
+  };
+
+  const clear = () => {
+    const params = new URLSearchParams(searchParams);
+
+    params.delete('sex');
```

Comment:
Also, I understand trying to get it in your own way, but as a suggestion: we have a SearchLink file and a helper function in utilities (getSearchWith) and they are already connected, so you could've used that and write much less (you would only need to pass sex: null, query: null ... in params for SearchLink. Again, it's only a suggestion of how it could be done in another way. Also this way you would not need to write handle centuries and handleSex

Reply:
Yeah I know,  just did not want to use a premaid solution, in order to train